### PR TITLE
syncthing: 1.1.4 -> 1.2.1

### PR DESCRIPTION
**Pull request by emmanuelrosa**
syncthing-gtk: add missing runtime dependencies

NixOS: fix syncthing-init NixOS test

###### Motivation for this change

The purpose of this change is to upgrade syncthing from 1.1.4 to 1.2.1.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - [X] syncthing-init
  - [X] syncthing-relay
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
  - [X] syncthing-tray
  - [X] qsyncthingtray
  - [X] syncthing-gtk
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/syncthing/default.nix [hunk 1]
```diff
@@ -3,14 +3,14 @@
 let
   common = { stname, target, postInstall ? "" }:
     buildGoPackage rec {
```
Comment (Lassulus) [inline pkgs/applications/networking/syncthing/default.nix]: If you use buildGoModule you can skip adding the deps.nix

**Comment by Lassulus** [comment]
@GrahamcOfBorg build syncthing

File: pkgs/applications/networking/syncthing/default.nix (hunk 1 above)
Comment (emmanuelrosa) [reply pkgs/applications/networking/syncthing/default.nix]: When I removed `goDeps = ./deps.nix` it failed to build.

Comment (Lassulus) [reply pkgs/applications/networking/syncthing/default.nix]: with these changes it build for me: https://gist.github.com/Lassulus/fc2cc45fc5caffa4e76885644c26e0fb

Comment (emmanuelrosa) [reply pkgs/applications/networking/syncthing/default.nix]: Ah, I see. Do you have a patch for this? I'll apply it and try it out. The patch above is missing the header, so Git won't apply it.

Comment (Lassulus) [reply pkgs/applications/networking/syncthing/default.nix]: ah sure: http://paste.krebsco.de/9mntuseY/+inline

File: pkgs/applications/networking/syncthing/deps.nix [hunk 2]
```diff
@@ -9,6 +9,15 @@
       sha256 = "1accmpl1llk16a19nlyy991fqrgfay6l53gb64hgmdfmqljdvbk7";
     };
   }
+  {
```
Comment (Lassulus) [inline pkgs/applications/networking/syncthing/deps.nix]: I guess you can remove this file now, since it's no longer needed

**Comment by Lassulus** [comment]
@GrahamcOfBorg test syncthing-init syncthing-relay

**Review by Lassulus — APPROVED** [review]

**Review by bachp — APPROVED** [review]
Sucessfully running on my server

**Comment by Infinisil** [comment]
The ofborg checks didn't succeed, which breaks master evaluation (in addition to another cause), both are fixed with above PR. Reviewers should look at these checks before merging

**Pull request merged**